apps/service_define/lib/define/event/store.ex

```diff
@@ -0,0 +1,85 @@
+defmodule Define.Event.Store do
+  require Logger
+  alias Define.DefinitionSerialization
+  alias Define.Model.{DataDefinitionView, ExtractView, LoadView, TransformView}
+  import Definition, only: [identifier: 1]
+
+  @instance Define.Application.instance()
+  @collection "definitions"
+
+  def update_definition(%Transform{} = data) do
+    update_definition_field(data, :transform, &to_transform_view/1)
+  end
+
+  def update_definition(%Load{} = data) do
+    update_definition_field(data, :load, &to_load_view/1)
+  end
+
+  def update_definition(%Extract{} = data) do
+    update_definition_field(data, :extract, &to_extract_view/1)
+  end
+
+  def update_definition(data) do
+    Logger.error(fn -> "Got unexpected data definition update: #{inspect(data)}" end)
+  end
+
+  defp update_definition_field(data, key, to_view_converter) do
+    get_or_create(data)
+    |> Map.put(key, to_view_converter.(data))
+    |> persist()
+  rescue
+    err ->
+      Logger.error(fn -> "Unable to process transform event: #{inspect(err)}" end)
+      discard()
```

Comment:
```suggestion
      :discard
```
I think we've taken the abstraction too far.

Reply:
Yeah, I don't care for this either.